crates/database/src/order_events.rs

```diff
@@ -67,6 +67,34 @@ pub async fn insert_order_event(
         .map(|_| ())
 }
 
+/// Insert a row into the `order_events` table if a previous event for the order
+/// `uid` exists.
+pub async fn insert_order_event_if_exists(
+    ex: &mut PgConnection,
+    event: &OrderEvent,
+) -> Result<(), sqlx::Error> {
+    const QUERY: &str = r#"
+        INSERT INTO order_events (
+            order_uid,
+            timestamp,
+            label
+        )
+        SELECT $1, $2, $3
+        WHERE EXISTS (
+            SELECT 1
+            FROM order_events
+            WHERE order_uid = $1
```

Comment:
The concept of this function seems pretty implicit.
Since you already do another query to determine whether an insert should actually happen couldn't you do this?
```sql
SELECT $1, $2, $3
        WHERE EXISTS (
            SELECT 1
            FROM orders
            WHERE uid = $1 AND class = 'market'
        )
```
That way we only need a single function `insert_event_if_market_order()`.

Reply:
As for now, we need to update the existing non-market orders until all the non-market orders are deleted by the periodic cleaning mechanism. `AND class = 'market'` means that we stop inserting events for all the existing non-market orders, which might not be an expected behavior.

Reply:
Nah, it's totally fine to completely stop inserting events for non-market orders. The `order_events` table only contains debugging data that is used for this [grafana query](https://g-0263500beb.grafana-workspace.eu-central-1.amazonaws.com/d/Osrh8t_4k/cow-slis?orgId=1&editPanel=8) which already ignores non-market orders.

Reply:
> Nah, it's totally fine to completely stop inserting events for non-market orders.

In that case, I need to change the implementation for the entire PR. Will open a separate one.